Modules/Module_Import/Import/xASL_imp_DetermineStructureFromSourcedata.m

```diff
@@ -240,73 +272,60 @@ function xASL_imp_PrintRun(x,thisSubject,thisVisit,thisRun)
     %% Preallocate space for (global) counts
     [x.importOverview.(sFieldName),x.importOverview.(sFieldName).(vFieldName)] = ...
         xASL_imp_PreallocateGlobalCounts(x.modules.import.nSubjects,x.importOverview.(sFieldName),x.importOverview.(sFieldName).(vFieldName));
-    
 
 end
 
-
+% -----------------------------------------------------------------
 %% Add fields of this subject/visit
+% -----------------------------------------------------------------
 function x = xASL_imp_thisSubjectVisit(x,sFieldName,vVisitIDs,vFieldName)
 
-
     % Visits (= sessions in BIDS)
     x = xASL_imp_AddVisitNames(x,sFieldName);
-    
-    
+   
     % Sessions (= run in BIDS)
     x = xASL_imp_AddSessionNames(x,sFieldName,vFieldName,vVisitIDs);
     
-    
     % Scans (= actual DICOM data)
     x = xASL_imp_AddScanNames(x,sFieldName,vFieldName,vVisitIDs);
-    
 
     % Sessions (= runs in BIDS)
     x = xASL_imp_AddSessions(x,sFieldName,vFieldName);
-
-
 end
 
-
+% -----------------------------------------------------------------
 %% Add visit names
+% -----------------------------------------------------------------
 function x = xASL_imp_AddVisitNames(x, sFieldName)
 
     if isempty(x.modules.import.imPar.visitNames)
         if isempty(x.importOverview.(sFieldName).visitIDs)
+			% In case that visit names are not filled, we name the visit by their number
             x.modules.import.imPar.visitNames = cell(x.importOverview.(sFieldName).nVisits,1);
             for iVisit=1:x.importOverview.(sFieldName).nVisits
-                x.modules.import.imPar.visitNames{iVisit} = sprintf('ASL_%g', iVisit);
+                x.modules.import.imPar.visitNames{iVisit} = xASL_num2str(iVisit);
             end
         else
             for iVisit=1:numel(x.importOverview.(sFieldName).visitIDs)
 				% Find the name id of the visit according to its number
-				idVisit = cellfun(@(y) regexp(y, x.importOverview.(sFieldName).visitIDs{iVisit}), x.modules.import.imPar.tokenVisitAliases(:,1), 'UniformOutput', false);
-				idVisit = find(cellfun(@(y) ~isempty(y), idVisit));
+				idVisit = cellfun(@(y) strcmp(y, x.importOverview.(sFieldName).visitIDs{iVisit}), x.modules.import.imPar.tokenVisitAliases(:,1), 'UniformOutput', false);
```

Comment:
This was programmed in the way that tokenVisitAliases are regular expressions, are you sure this still works for that case? They should be regular expressions, like all aliases are.

Reply:
Oh. That was a useful comment. I have to change this. I changed it because regexp cannot handle looking for empty strings, so I have to do this manually one-by-one

Reply:
I would just use a single regexp. Regexp is able to handle empty strings if you always add `|)` or `(|` by default right? You can resolve.